test/powershell/engine/Remoting/InvokeCommandRemoteDebug.Tests.ps1

```diff
@@ -0,0 +1,200 @@
+        {
+            args.HandleInternally = true;
+            RunspaceDebugProcessingCount++;
+        }
+        private void HandleCancelProcessRunspaceDebug(object sender, EventArgs args)
+        {
+            RunspaceDebugProcessCancelled = true;
+        }
+        public TestDebugger(Runspace runspace)
+        {
+            _runspace = runspace;
+            _runspace.Debugger.DebuggerStop += HandleDebuggerStop;
+            _runspace.Debugger.ProcessRunspaceDebug += HandleProcessRunspaceDebug;
+            _runspace.Debugger.CancelProcessRunspaceDebug += HandleCancelProcessRunspaceDebug;
+        }
+        public void Release()
+        {
+            if (_runspace == null) { return; }
+            _runspace.Debugger.DebuggerStop -= HandleDebuggerStop;
+            _runspace.Debugger.ProcessRunspaceDebug -= HandleProcessRunspaceDebug;
+            _runspace.Debugger.CancelProcessRunspaceDebug -= HandleCancelProcessRunspaceDebug;
+        }
+
+        public override DebuggerCommandResults ProcessCommand(PSCommand command, PSDataCollection<PSObject> output) { return null; }
+        public override void SetDebuggerAction(DebuggerResumeAction resumeAction) { }
+        public override DebuggerStopEventArgs GetDebuggerStopArgs() { return null; }
+        public override void StopProcessCommand() { }
+    }
+}
+'@
+
+Describe "Invoke-Command remote debugging tests" -Tags 'Feature' {
+
+    BeforeAll {
+
+        if (!$IsWindows)
+        {
+            $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+            $PSDefaultParameterValues["it:skip"] = $true
+            return
+        }
+
+        $sb = [scriptblock]::Create(@'
+        "Hello!"
+'@)
+
+        Add-Type -TypeDefinition $typeDef -ReferencedAssemblies "System.Globalization","System.Management.Automation"
+
+        $dummyHost = [TestRunner.DummyHost]::new()
+        [runspace] $rs = [runspacefactory]::CreateRunspace($dummyHost)
+        $rs.Open()
+        $dummyHost._runspace = $rs
+
+        $testDebugger = [TestRunner.TestDebugger]::new($rs)
+
+        [runspace] $rs2 = [runspacefactory]::CreateRunspace()
+        $rs2.Open()
+
+        [powershell] $ps = [powershell]::Create()
+        $ps.Runspace = $rs
+
+        [powershell] $ps2 = [powershell]::Create()
+        $ps2.Runspace = $rs2
+
+        $remoteSession = New-RemoteRunspace
+    }
+
+    AfterAll {
+
+        if (!$IsWindows)
+        {
+            $global:PSDefaultParameterValues = $originalDefaultParameterValues
+            return
+        }
+
+        if ($testDebugger -ne $null) { $testDebugger.Release() }
+        if ($ps -ne $null) { $ps.Dispose() }
+        if ($ps2 -ne $null) { $ps2.Dispose() }
+        if ($rs -ne $null) { $rs.Dispose() }
+        if ($rs2 -ne $null) { $rs2.Dispose() }
+        if ($remoteSession -ne $null) { Remove-PSSession $remoteSession -ErrorAction SilentlyContinue }
+    }
+
+    It "Verifies that asynchronous Invoke-Command -RemoteDebug is ignored" {
```

Comment:
It would be better:
```powershell
It "Verifies that asynchronous 'Invoke-Command -RemoteDebug' is ignored" { 
```
It would be good to fix headers below similarly.